### ISSUE TITLE
amule: segmentation fault

### ISSUE THREAD

**Issue by eatradish** [AOSC-Dev/aosc-os-abbs#2076]
Bug description
-----------------------

Segmentation fault

```
➜  temp gdb amule
...
Reading symbols from amule...(no debugging symbols found)...done.
(gdb) run
Starting program: /usr/bin/amule 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Detaching after vfork from child process 39830]
 2020-01-16 19:55:02: Initialising aMule 2.3.2 compiled with wxGTK2 v3.0.4
 2020-01-16 19:55:02: Checking if there is an instance already running...
 2020-01-16 19:55:02: No other instances are running.
[Detaching after vfork from child process 39832]
[Detaching after vfork from child process 39834]
[Detaching after vfork from child process 39836]
[Detaching after vfork from child process 39838]
[Detaching after vfork from child process 39840]
[Detaching after vfork from child process 39844]
[Detaching after vfork from child process 39848]
[Detaching after vfork from child process 39852]
 2020-01-16 19:55:02: ListenSocket: Ok.
[New Thread 0x7fffe7256700 (LWP 39859)]
[New Thread 0x7fffe6a55700 (LWP 39860)]
[New Thread 0x7fffe61c8700 (LWP 39861)]
[Detaching after vfork from child process 39862]
[New Thread 0x7fffe53ae700 (LWP 39864)]
[New Thread 0x7fffe4bad700 (LWP 39865)]
 2020-01-16 19:55:03: Loading temp files from /home/sakiiily/.aMule/Temp.
 2020-01-16 19:55:03: All PartFiles Loaded.
[Thread 0x7fffe4bad700 (LWP 39865) exited]
[New Thread 0x7fffe4bad700 (LWP 39866)]
[Thread 0x7fffe4bad700 (LWP 39866) exited]
[New Thread 0x7fffcbfff700 (LWP 39867)]
[New Thread 0x7fffbde8b700 (LWP 39868)]
[New Thread 0x7fffbd68a700 (LWP 39869)]
[New Thread 0x7fffbce89700 (LWP 39870)]
[Thread 0x7fffbd68a700 (LWP 39869) exited]

Thread 1 "amule" received signal SIGSEGV, Segmentation fault.
0x00007ffff74e5c07 in wxWindow::DoClientToScreen(int*, int*) const () from /usr/lib/libwx_gtk3u_core-3.0.so.0
(gdb) bt
#0  0x00007ffff74e5c07 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#1  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#2  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#3  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#4  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#5  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#6  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#7  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#8  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#9  0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#10 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#11 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#12 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#13 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#14 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#15 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#16 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#17 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#18 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#19 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#20 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#21 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#22 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#23 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#24 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#25 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#26 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#27 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#28 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#29 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#30 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#31 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#32 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#33 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#34 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#35 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#36 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#37 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#38 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#39 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#40 0x00007ffff74e5ce0 in wxWindow::DoClientToScreen(int*, int*) const () at /usr/lib/libwx_gtk3u_core-3.0.so.0
...
```

Reproducing steps
---------------------------

1. Open amule
2. Segmentation fault

Update channel(s)
--------------------------

Testing-Proposed

Package and version
------------------------------

```
➜  temp dpkg -s amule
Package: amule
Status: install ok installed
Section: net
Installed-Size: 24952
Maintainer: sakiiily <sakiiily@aosc.io>
Architecture: amd64
Version: 1:2.3.2-6
Depends: binutils (>= 2.33.1), wxgtk (>= 2.8.12.1-8), libgd (>= 2.2.5-5), geoip-api-c (>= 1.6.12), libupnp (>= 1.6.20), crypto++ (>= 8.2.0), libwebp (>= 1.0.3), wx-base (>= 1)
Description: An eMule-like client for ed2k network
```

**Comment by LionNatsu** [comment]
Looks like a stack overflow happened in wx.

**Comment by MingcongBai** [comment]
Fixed in 12c7794ba9f909c9efcd66d28724ab4e71cc25ca, it seems that aMule is still incompatible with the GTK+ 3 wx runtime. Closing.
